syncd/FlexCounter.cpp

```diff
@@ -1934,17 +2090,67 @@ void FlexCounter::updateSupportedPriorityGroupCounters(
     }
 }
 
-void FlexCounter::updateSupportedRifCounters(
-        _In_ sai_object_id_t rifRid)
+void FlexCounter::updateSupportedPriorityGroupCounters(
+        _In_ sai_object_id_t priorityGroupRid,
+        _In_ const std::vector<sai_ingress_priority_group_stat_t> &counterIds)
 {
     SWSS_LOG_ENTER();
 
-    if (m_supportedRifCounters.size())
+    m_supportedPriorityGroupCounters.clear();
+
+    /* Query SAI supported priority group counters */
+    sai_status_t status = querySupportedPriorityGroupCounters(priorityGroupRid);
+    if (status != SAI_STATUS_SUCCESS)
     {
-        return;
+        /* Fallback to legacy approach */
+        getSupportedPriorityGroupCounters(priorityGroupRid, counterIds);
+    }    
+}
+
+sai_status_t FlexCounter::querySupportedRifCounters(
+        sai_object_id_t rifRid)
+{
+    sai_stat_capability_list_t stats_capability;
+    stats_capability.count = 0;
+    stats_capability.list = nullptr;
+
+    /* First call is to check the size needed to allocate */
+    sai_status_t status = m_vendorSai->queryStatsCapability(
+        rifRid, 
+        SAI_OBJECT_TYPE_ROUTER_INTERFACE, 
+        &stats_capability);
+
+    /* Second call is for query statistics capability */
+    if (status == SAI_STATUS_BUFFER_OVERFLOW)
+    {
+        std::vector<sai_stat_capability_t> statCapabilityList(stats_capability.count);
+        stats_capability.list = statCapabilityList.data();
+        status = m_vendorSai->queryStatsCapability(
+            rifRid, 
+            SAI_OBJECT_TYPE_ROUTER_INTERFACE, 
+            &stats_capability);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_INFO("Unable to get routed port supported counters for %s", sai_serialize_object_id(rifRid).c_str());
+        }
+        else
+        {
+            for (uint32_t i = 0; i < stats_capability.count; i++)
```

Comment:
Why not to use for range loop since you anyway use a vector?
```suggestion
            for (auto capability: statCapabilityList)
```

Reply:
Fixed